### DISCUSSION TITLE
Shopping Cart

### DISCUSSION THREAD

**Discussion by jeffolmstead** [OrchardCMS/OrchardCore.Commerce#67] [General] [answered]
@bleroy I see you have a good start on the back end of the shopping cart (e.g. controller / persistence). Were you planning on implementing the front end via similar scripts to what you had in Orchard (non-Core)? I might have an opening to port them over if that is where you were leaning. Just let me know.

**Comment by bleroy** [comment, marked as answer]
Yes, that’s the idea. I implemented a scriptless version but if you want to port the script, that’s great. The current behavior should still work as a fallback.

**Comment by jeffolmstead** [comment]
@bleroy can you add me back on to OrchardCore.Commerce git repository as I was bumped when the root project (e.g. OrchardCMS) made the change to require two factor authentication and mine wasn't set up when they made that change. It is set up now.

**Reply by bleroy** [reply]
Sure. Done. I invited you back to the org with the rights you had before. LMK if you need me to do something specific on this repo.

**Comment by mwpowellhtx** [comment]
Chiming in here, I think we have a requirement, potentially, to support perhaps blogs attached to _cart_ and/or order _items_, in order to capture _pre/post-sale discussion_, _agreement to terms of service_, _excahnge of assets_, and this sort of thing. High level, I am thinking in terms of _OC Workflow_, would that be possible to support those phase lines in terms of workflow events. And, for our purpose, to associate _Commerce_ _Cart_/_Order_ _content items_ with _blog content items_, along these lines.

**Reply by bleroy** [reply]
Yes. The main reason for modeling the checkout process as a workflow is so it can accommodate for custom business models. As long as your processes are exposed as workflow activities, you should be able to integrate them.

**Reply by mwpowellhtx** [reply]
_Attached_ is a strong word, perhaps, _associated with_, along the lines of _relationship between_ `blog content item` and `cart/order content item`.
